src/explosion.h

```diff
@@ -55,15 +56,20 @@ struct explosion_data {
 // handles explosion related functions
 namespace explosion_handler
 {
-/** Create explosion at p of intensity (power) with (shrapnel) chunks of shrapnel.
+using queued_explosion = std::pair<tripoint, explosion_data>;
+static std::vector<queued_explosion> _explosions;
+
+/** Queue an explosion at p of intensity (power) with (shrapnel) chunks of shrapnel.
     Explosion intensity formula is roughly power*factor^distance.
-    If factor <= 0, no blast is produced */
+    If factor <= 0, no blast is produced
+    The explosion won't actually occur until process_explosions() */
 void explosion(
     const tripoint &p, float power, float factor = 0.8f,
     bool fire = false, int casing_mass = 0, float frag_mass = 0.05
 );
 
 void explosion( const tripoint &p, const explosion_data &ex );
+void _explosion( const tripoint &p, const explosion_data &ex );
```

Comment:
A different name maybe? It is a bit confusing now.

Reply:
I changed it to `_make_explosion`. Let me know if you have a better idea (or feel free to change it).